### DISCUSSION TITLE
Inconsitent use record field type separator

### DISCUSSION THREAD

**Discussion by kuchta** [flix/flix#6564] [Ideas]
The rest (at least what I've came across) of the language uses colon as field type separator, but records use equal sign.

**Comment by magnus-madsen** [comment]
Do you have an example? 

I believe we use `:` for type and kind ascriptions, and we use for `=` for associating a label with a type. Is that what you mean?

**Comment by kuchta** [comment]
Hi @magnus-madsen . Yes, I've meant e.g.

```
def area(r: {x = Int32, y = Int32}): Int32 = r.x * r.y
```
I think that
```
def area(r: {x : Int32, y : Int32}): Int32 = r.x * r.y
```
would be more consistent. I've just started to learn the language and I really love how consistent the syntax otherwise is ❤️

**Comment by magnus-madsen** [comment]
I think we used to have that, but it caused problems. @mlutze Do you remember?

>  I've just started to learn the language and I really love how consistent the syntax otherwise is ❤️

Thanks! :)

**Comment by mlutze** [comment]
> I think we used to have that, but it caused problems. @mlutze Do you remember?

This was due to the introduction of (experimental) set types. `{ a: X }` could be a record type with field `a` and type `X`, or it could be a set type containing member `a` with kind `X`.

**Comment by magnus-madsen** [comment]
Also note that this is consistent: `{ x = 123, y = 456 }` has type `{ x: Int32, y: Int32 }`.

We cannot use `:` at the expression level because it is a type annotation.

**Comment by kuchta** [comment]
Also reading section [Named Parameters with Records](https://doc.flix.dev/records.html#named-parameters-with-records) I think that introducing syntactic sugar in the form
```
translate(from = English, to = French, "Where is the library?")
```
for calling function with 
```
translate({from = English}, {to = French}, "Where is the library?")
```
could cause problems if you decide to implement "real" named parameters which IMHO doesn't need any special function anotation like Swift e.g. uses. In Nim it is just matter of calling convention [Nim: Procedures](https://nim-lang.org/docs/manual.html#procedures)

**Comment by magnus-madsen** [comment]
You might be interested in Principle 36 in https://flix.dev/paper/onward2022.pdf

I think it describes how Flix' approach to named parameters is better than other approaches.

**Comment by kuchta** [comment]
Well, I'm not sure if you haven't solved the problem you basically introduced by using records for named parameters. But I don't know how to implement named (not positional) partial application in a curried language, though. It just seems to me, that if the curried language has first class support for named parameters, it could just partially apply them...

**Comment by kuchta** [comment]
Regarding the arguments:
> But labelled arguments, in their typical implementation, has at least two downsides:
• they are optional; hence programmers may forget to use them, and
• they are typically not part of the type system, hence they do not work with first-class functions.

- The first one is definitely true, but I'm not sure if it's good practice to force the programmer to wrap the arguments in some structure with different performace characteristics just becase two parameters happen to use the same type which can be also remedied by using `distict type alias` if really needed
- Unfortunatelly I don't really understand the second one

**Comment by mlutze** [comment]
I believe the second point is in reference to the issues outlined here with OCaml's named operators: https://v2.ocaml.org/manual/lablexamples.html#s:label-inference

In short, type inference breaks with named parameters in the OCaml style.

I agree that some built-in rules or use of a `distinct type alias` could work as well, and might be more appropriate, since there is a semantic difference between a singleton record and a named argument, even if they are isomorphic.
